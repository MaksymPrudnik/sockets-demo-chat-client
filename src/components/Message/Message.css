.message-li {
    display: flex;
    flex-direction: column;
    align-self: start;
    max-width: 80%;
    word-wrap: break-word;
    margin: 0.3rem;
}

.message-li-right {
    align-self: flex-end;
}

.message-author {
    text-align: left;
}

.message-body {
    padding: 0.2rem;
    border: 0.5px solid #aaa;
    border-radius: 0.5rem;
}

.message-date {
    font-size: 0.8rem;
    color: rgb(150, 150, 150);
    font-weight: 300;
}