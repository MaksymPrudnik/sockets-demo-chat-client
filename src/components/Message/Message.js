import React from 'react';
import './Message.css';

const Message = ({message, username}) => {
    return <li className={`message-li message-li-${username === message.username && 'right'}`}>
        { username !== message.username && <span className='message-author'>{message.username}</span> }
        <span className='message-body'>{message.message}</span>
        <span className='message-date'>{message.createdAt}</span>
    </li>
}

export default Message;