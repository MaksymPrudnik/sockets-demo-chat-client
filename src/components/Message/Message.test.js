import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message';

describe('<Message />', () => {
    const props = {
        username: 'maksymprudnik',
        message: {
            username: 'max',
            body: 'hi',
            img: 'https://sockets-chat-maksymprudnik.s3.eu-central-1.amazonaws.com/37-2.png',
            createdat: 'some/date'
        }
    }
    const wrapper = shallow(<Message {...props} />)
    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    it('should display message author optionaly', () => {
        expect(wrapper.find('span.message-author').length).toEqual(1);
        const equalUsernamesWrapper = shallow(<Message username={props.username} message={{...props.message, username: props.username}} />)
        expect(equalUsernamesWrapper.find('span.message-author').length).toEqual(0);
    })

    it('should display message image optionaly', () => {
        expect(wrapper.find('img').length).toEqual(1);
        const noImageWraper = shallow(<Message username={props.username} message={{...props.message, img: ''}} />)
        expect(noImageWraper.find('img').length).toEqual(0);
    })

    it('should display message body optionaly', () => {
        expect(wrapper.find('span.message-body').length).toEqual(1);
        const noBodyWraper = shallow(<Message username={props.username} message={{...props.message, body: ''}} />)
        expect(noBodyWraper.find('span.message-body').length).toEqual(0);
    })

})