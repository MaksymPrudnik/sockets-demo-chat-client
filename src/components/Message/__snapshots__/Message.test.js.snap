// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should match the snapshot 1`] = `"<li class=\\"message-li message-li-false\\"><span class=\\"message-author\\">max</span><img src=\\"https://sockets-chat-maksymprudnik.s3.eu-central-1.amazonaws.com/37-2.png\\" alt=\\"added content\\" class=\\"message-image\\"/><span class=\\"message-body\\">hi</span><span class=\\"message-date\\">some/date</span></li>"`;
